modules/aggregation-layers/src/utils/resource-utils.js

```diff
@@ -29,6 +29,29 @@ export function getFloatTexture(gl, opts = {}) {
   return texture;
 }
 
+export function getUByteTexture(gl, opts = {}) {
+  const {
+    width = 1,
+    height = 1,
+    data = null,
+    unpackFlipY = true,
+    parameters = DEFAULT_PARAMETERS
+  } = opts;
```

Comment:
As an internal helper function, I'd like to keep these very concise and not too "smart", as in as few defaults/assumptions as possible. This function constructs a texture with a specific type. All other parameter defaults are either redundant because they are the same defaults as used by `Texture2D`, or error-prone because they are not implied by the name of the function. To keep this helper flexible, you can simply pass through the opts like

```js
return new Texture2D(gl, {type: GL.UNSIGNED_BYTE, ...opts});
```

Comparing to your implementation, it is easy to test (only one branch), and has minimal impact on bundle size.

Reply:
Updated, keeping `getFloatTexture` for now, as it is used in aggregator, will clean that up in a different PR.